### PR TITLE
[DON'T MERGE] Update to 0.7.0-git

### PR DESCRIPTION
**Pull request by RobLoach**
This updates to mgba 0.7.0-git, and resolves the git conflicts between the two git repositories.

### CODE REVIEW

**Comment by RobLoach** [comment]
Thanks @sergiobenrocha2 :+1: ... Opened as a way going forwards for when 0.7.0 comes out. Feel free to test as needed.

**Comment by sergiobenrocha2** [comment]
You should rollback to 34474ff first I guess, then merge stuff, then merge it against a different branch, not master.

It was what I did in the last update, the current master was actually 0.6-wip branch or something like that. Then I pushed the old master to 0.5.x branch

**Comment by sergiobenrocha2** [comment]
Here: https://github.com/libretro/mgba/tree/0.7-wip

File: src/platform/libretro/libretro.c [hunk 1]
```diff
@@ -146,7 +146,7 @@ void retro_get_system_info(struct retro_system_info* info) {
 #ifndef GIT_VERSION
 #define GIT_VERSION ""
 #endif
-	info->library_version = "0.6.1" GIT_VERSION;
+	info->library_version = "0.7.0-git" GIT_VERSION;
```
Comment (sergiobenrocha2) [inline src/platform/libretro/libretro.c]: That should be 0.7-git or 0.7-wip, there's no 0.7.0 release yet

**Comment by sergiobenrocha2** [comment]
Seems endrift was working in 2 different branch I think (before 0.6.0 release), our commits does not match upstream master, even rolling back. Should we use upstream master branch as base and apply a libretro patch? This way should be much more cleaner i think

I tested here, it's working. It's up to you guys I think

**Comment by endrift** [comment]
If you do use upstream make sure to change the PS3 to use `-std=gnu99` instead of `-std=c99` or else it will have tons of build failures.

**Comment by RobLoach** [comment]
> Should we use upstream master branch as base and apply a libretro patch?

Always desirable to push fixes upstream. The only big change that we'd want here are the local dependencies. I'd even go as far to suggest moving the Makefile upstream to https://github.com/mgba-emu/mgba/tree/master/src/platform/libretro .

> change the PS3 to use -std=gnu99 instead of -std=c99

Thanks for the tip. Made a PR over at https://github.com/libretro/mgba/pull/77 .

**Comment by endrift** [comment]
The reason I don't want the Makefile upstream (historically) is to prevent from confusing users who see a makefile and try to use it instead of using CMake for building the main project, plus any potential drift between the CMake buildsystem and the libretro makefile.

That said, I may reconsider, but the upstream and downstream libretro cores have diverged a bit too. There are some additions I'm unhappy with (e.g. turbo buttons being mapped explicitly to buttons instead of having a generic frontend turbo button interface, exposing one of the two main memory regions as *the* main memory region, with no real way for a user to know, etc.)

I'm not opposed to attempting to reconcile the cores; in fact I'd like to see things drift LESS, not more, but I'd still have to consider if I want to accommodate all of the things you'd want to have changed.

**Pull request closed without merging**